### PR TITLE
Inventory management improvements.

### PR DESCRIPTION
**Pull request by The-Albinobigfoot**
- No more pixel hunting for obnoxiously small sprites. Now you can just click anywhere in an item's  inventory slot and equip/activate it.
- Works for storage containers, inventory, and borg/mommi modules.

Don't thank me; most of this was ported, line for line.
Brought to you by: RemieRichards of TG and Tastyfish of Paradise.

### CODE REVIEW

**Comment by SkowronX** [comment]
:+1: Very good yes

**Comment by Konater** [comment]
:eggplant:

**Comment by clusterfack** [comment]
MMMM I was going to port this but I completely forgot about it, haha

Anyway did you remember to test it

**Comment by The-Albinobigfoot** [comment]
Just a bit to make sure that it worked as intended, the borg's module functionality especially, as TG's borg set-up is a bit different than ours. Everything seems in working order, and I noticed no blatantly aberrant behavior.

**Comment by ComicIronic** [comment]
Does this just let you click the storage hud box instead of the item?

**Comment by RemieRichards** [comment]
@ComicIronic objects in an inventory (any "inventory") are set to mouse_opacity 2, which is just a Byond var to ignore transparency on all pixels in the sprite, so all the transparent pixels around the sprite become valid for clicks.

If you don't know,
`mouse_opacity = 0` -> Can never click it
`mouse_opacity = 1` -> The default, click where their are atleast 1 alpha pixels
`mouse_opacity = 2` -> Can click anywhere within it's icon size (32x32, 64x64 etc.)

**Comment by PJB3005** [comment]
IMO it would be better if the icon slot handles clicking instead of using mouse opacity, yes it's more code but I think it's cleaner.

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -495,6 +495,10 @@ var/global/obj/screen/fuckstat/FUCK = new
 	var/obj/item/W = get_active_hand()
 	if(istype(W))
 		equip_to_slot_if_possible(W, slot)
+	else
+		W = get_item_by_slot(slot)
+		if(W)
+			W.attack_hand(src)
```
Comment (The-Albinobigfoot) [inline code/modules/mob/mob.dm:499]: @PJB3005 You mean something like this?

Comment (PJB3005) [inline code/modules/mob/mob.dm:499]: Oh, yes I do.

I guess this PR just does both, then.

Comment (The-Albinobigfoot) [inline code/modules/mob/mob.dm]: You know, it just occurred to me that I didn't test this with just that bit there. When I get home later I'll test it.

**Comment by Kurfursten** [comment]
How does this handle quickswapping? I don't believe TG has quickswapping and it's one of Pomf's favorites. I know I can't get used to not having it.

Basically the idea is if you click the slot behind an equipped item with an item that could go in that slot, you switch the item in your hand and slot (without dropping things like your pockets and ID if it was a suit)

**Comment by PJB3005** [comment]
See this is why mouse opacity is a bad idea.

**Comment by The-Albinobigfoot** [comment]
I'll test that too when i get the chance. Also, I forgot about mommies, fuck me.

**Comment by The-Albinobigfoot** [comment]
Good news, friends, it seems to work flawlessly with quickswap, both using the 'equip' shortcut, and manually clicking on the desired slot.
The items retain their opacity as well, no strangeness detected overall.

**Comment by The-Albinobigfoot** [comment]
> hand refactor was merged

FUCK

**Comment by The-Albinobigfoot** [comment]
Send help.

**Comment by Shadowmech88** [comment]
> giving up

For shame.

**Comment by The-Albinobigfoot** [comment]
I'm just going to try and figure it out before I make a complete mess of the PR.

**Comment by The-Albinobigfoot** [comment]
https://github.com/The-Albinobigfoot/vgstation13/commit/79dce422f07b27d041068fe19a373ef79e95e5d1
Worked perfectly fine there.
It just doesn't make any god damn sense.

**Comment by The-Albinobigfoot** [comment]
> can't re-open when you've force-pushed.

**Comment by PJB3005** [comment]
Just make a new PR, it's not a bother.

**Pull request closed without merging**